File: Native/Android/LunarConsole/lunarConsole/src/main/java/spacemadness/com/lunarconsole/utils/CompositeDisposable.java
package spacemadness.com.lunarconsole.utils;

import java.util.ArrayList;
import java.util.List;

import spacemadness.com.lunarconsole.core.Disposable;

public class CompositeDisposable implements Disposable {
    private final List<Disposable> disposables;

    public CompositeDisposable() {
        disposables = new ArrayList<>();
    }

    public void add(Disposable disposable) {
        disposables.add(disposable);
    }

    @Override
    public void dispose() {
        for (Disposable disposable : disposables) {
            disposable.dispose();
        }
        disposables.clear();
    }
}

File: Native/Android/LunarConsoleNew/lunarConsole/src/main/java/spacemadness/com/lunarconsole/core/CompositeDisposable.kt
package spacemadness.com.lunarconsole.core

class CompositeDisposable : Disposable {
    private val disposables = mutableListOf<Disposable>()

    fun add(disposable: Disposable): CompositeDisposable {
        disposables.add(disposable)
        return this
    }

    override fun dispose() {
        disposables.forEach { it.dispose() }
    }
}

fun Disposable.addTo(disposables: CompositeDisposable) {
    disposables.add(this)
}